backend/xivdpscalc/pps/sch.py

```diff
@@ -12,6 +17,71 @@ class SchPps(HealerPps):
     bio_potency = 70
     ed_potency = 100
 
+    def get_total_potency_variable_time(self, sim_length: ElapsedTime, character_stats: Character, rotation: SchRotation, caster_tax: float = 0.1) -> float:
+        """
+        Get the total potency in sim_length seconds
+        rotation is a SchRotation object
+        todo: support dot damage calculation
+        todo: how long is animation lock actually?  Hard coded here for 0.75, have no idea if that's true
+        There are some nuances to how this works:
+            At present, this sim does not calculate dot damage
+            If an action with an active cooldown is selected, the time jumps forward to the cooldown's expiry
+            This simulator does not enforce resource requirements - ED can be used even with no aetherflow
+        """
+        animation_lock = 0.75
+
+        total_potency = 0
+        current_time = 0
+
+        short_gcd = character_stats.get_gcd()
+
+        next_active = {action: 0 for action in SchAction}
+        active_effects = defaultdict(lambda: -1)
```

Comment:
I think we need a typing here, since mypy isn't happy with it. I'd suggest [typing.DefaultDict](https://docs.python.org/3/library/typing.html#typing.DefaultDict), but apparently `collections.DefaultDict` is appropriate for typing